django/core/checks/urls.py

```diff
@@ -28,6 +28,41 @@ def check_resolver(resolver):
         return []
 
 
+@register(Tags.urls)
+def check_url_namespaces_unique(app_configs, **kwargs):
+    """
+    Checks that all URL namespaces used in applications are unique.
+    """
+    if not getattr(settings, 'ROOT_URLCONF', None):
+        return []
+
+    from django.urls import get_resolver
+    resolver = get_resolver()
+
+    all_namespaces = _load_all_namespaces(resolver)
+    for namespace in all_namespaces:
+        if all_namespaces.count(namespace) > 1:
+            return [Warning(
```

Comment:
you should accumulate all warnings into a list and return that at the end of the function. At the moment this returns early, if there are two non-unique names in use, only the first one will get a warning. If the user fixes that, they run the checks again, and get a new warning about the second non-unique name. Not a great workflow if there are many to fix.